openedx/features/badging/models.py

```diff
@@ -18,3 +23,79 @@ class Badge(models.Model):
 
     def __unicode__(self):
         return self.name
+
+    @classmethod
+    def get_remaining_badges(cls, user_id, community_id):
+        course_id = get_course_id_by_community_id(community_id)
+
+        if course_id is not CourseKeyField.Empty:
```

Comment:
lets make formal methods for this

Reply:
Outdated, code changed as per suggested query in the comment [above](https://github.com/philanthropy-u/edx-platform/pull/1191/#discussion_r328612180)